consensus/spos/consensusState.go

```diff
@@ -64,6 +64,15 @@ func NewConsensusState(
 	return &cns
 }
 
+// ResetConsensusRoundState method resets all the consensus round data (except messages received)
+func (cns *ConsensusState) ResetConsensusRoundState() {
+	cns.RoundCanceled = false
+	cns.ExtendedCalled = false
+	cns.WaitingAllSignaturesTimeOut = false
+	cns.ResetRoundStatus()
+	cns.ResetRoundState()
+}
+
 // ResetConsensusState method resets all the consensus data
 func (cns *ConsensusState) ResetConsensusState() {
```

Comment:
we can call `ResetConsensusRoundState` in `ResetConsensusState` and reduce duplicated code